Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationFailureRepository.cs

```diff
@@ -0,0 +1,306 @@
+    /// its child <see cref="IBusinessObject"/> rows and
+    /// all of the <see cref="IBocColumnDefinitionWithValidationSupport"/> cells inside those rows.
+    /// </summary>
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledListValidationFailures (bool markAsHandled)
+    {
+      if (_listValidationFailures == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = new List<BusinessObjectValidationFailure>();
+      foreach (var handleableRowFailure in _listValidationFailures.Where(e => !e.Handled))
+      {
+        failures.Add(handleableRowFailure.Failure);
+        if (markAsHandled)
+          handleableRowFailure.MarkAsHandled();
+      }
+
+      return failures;
+    }
+
+    /// <summary>
+    /// Returns a collection of all <see cref="BusinessObjectValidationFailure"/>s belonging to a <see cref="BocList"/>.
+    /// </summary>
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetAllUnhandledListValidationFailures (bool markAsHandled)
+    {
+      if (_listValidationFailures == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = new List<BusinessObjectValidationFailure>();
+      foreach (var handleableRowFailure in _listValidationFailures.Where(e => !e.Handled))
+      {
+        failures.Add(handleableRowFailure.Failure);
+        if (markAsHandled)
+          handleableRowFailure.MarkAsHandled();
+      }
+
+      if (_rowValidationFailures != null)
+      {
+        foreach (var failureContainer in _rowValidationFailures.Values)
+        {
+          failureContainer.GetAllFailures(failures, false);
+        }
+      }
+
+      return failures;
+    }
+
+    /// <summary>
+    /// Returns a collection of all <see cref="BusinessObjectValidationFailure"/>s belonging to an <see cref="IBusinessObject"/> row and
+    /// all of the <see cref="IBocColumnDefinitionWithValidationSupport"/> cells inside that row.
+    /// </summary>
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledRowValidationFailures (IBusinessObject rowObject, bool markAsHandled)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rowObject), rowObject);
+
+      var failureContainer = GetRowFailureContainer(rowObject);
+      if (failureContainer == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = new List<BusinessObjectValidationFailure>();
+      failureContainer.GetRowFailures(failures, markAsHandled);
+
+      return failures;
+    }
+
+    /// <summary>
+    /// Returns a collection of all <see cref="BusinessObjectValidationFailure"/>s belonging to an <see cref="IBusinessObject"/> row.
+    /// </summary>
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetAllUnhandledRowValidationFailures (IBusinessObject rowObject, bool markAsHandled)
```

Comment:
naming: specificity - I think actually the naming might be inverse. Methods named "All" would return the failures for the  context object (e.g. the List or the Row) and all dependent objects (e.g. the Cells of the Row or the Rows of the List). The termn "All" is used for deciding if we want all failures or just some. Here, we also have failure catorization (e.g. the Row-asociated failures). I'm not sure if we don't want longer names that specify what type of failures are returned. Right now "GetAllUnhandledRowValidationFailures " means to me "row failures for all rows" and not "failures for this row including the cells". It's very possilbe, we always only want to include all nested failures when showing "unhandled failures" which makes the "all" naming redundant.